pyqtgraph/graphicsItems/ImageItem.py

```diff
@@ -197,17 +246,42 @@ def setOpts(self, update=True, **kargs):
             self.menu = None
         if 'autoDownsample' in kargs:
             self.setAutoDownsample(kargs['autoDownsample'])
+        if 'rect' in kargs:
+            self.setRect(kargs['rect'])
         if update:
             self.update()
 
-    def setRect(self, rect):
-        """Scale and translate the image to fit within rect (must be a QRect or QRectF)."""
+    def setRect(self, *args):
+        """
+        setRect(rect) or setRect(x,y,w,h)
+        
+        Sets translation and scaling of this ImageItem to display the current image within the rectangle given
+        as ``QtCore.QRect`` or ``QtCore.QRectF`` `rect`, or described by parameters `x, y, w, h`, defining starting 
+        position, width and height.
+
+        This method cannot be used before an image is assigned.
+        See the :ref:`examples <ImageItem_examples>` for how to manually set transformations.
+        """
+        rect = None
+        if len(args) == 4:
```

Comment:
probably should do `if len(args) in (2, 4)` as there are some valid QRectF constructors involving combinations of QPointF and QSizeF.... seems like a lot of work to check user-input in this case tho...

Reply:
```python
rect = None
if isinstance(args, (QRect, QRectF):
    rect = args
elif hasattr(args, '__len__'):
    arg = args[0]
    rect = QRectF(*arg)
else:
    rect = QRectF(*args)
```

This should keep working w/ all compatible QRectF initializers.

Reply:
The idea is not primarily to construct a QRectF, but rather to set the display rectangle directly from float values. But the code seemed easiest to read for building up the QRectF.

How's this for a minimalistic approach?
```python
if isinstance(args, (QtCore.QRectF, QtCore.QRect)):
            rect = args # use QRectF or QRect directly
        else:
            if hasattr(args[0],'__len__'):
                args = args[0] # promote tuple or list of values
            rect = QtCore.QRectF( *args ) # QRectF(x,y,w,h), but also accepts other initializers
```
That either produces a suitable rectangle, or results in a quite readable stack trace. This one happens for giving two float values:
```
File "C:\e\prog\pg_dev\pyqtgraph\pyqtgraph\graphicsItems\ImageItem.py", line 270, in setRect
    rect = QtCore.QRectF( *args ) # QRectF(x,y,w,h), but also accepts other initializers
TypeError: arguments did not match any overloaded call:
  QRectF(): too many arguments
  QRectF(Union[QPointF, QPoint], QSizeF): argument 1 has unexpected type 'int'
  QRectF(Union[QPointF, QPoint], Union[QPointF, QPoint]): argument 1 has unexpected type 'int'
  QRectF(float, float, float, float): not enough arguments
  QRectF(QRect): argument 1 has unexpected type 'int'
  QRectF(QRectF): argument 1 has unexpected type 'int'
```